src/gpu/d3d11/SDL_gpu_d3d11.c

```diff
@@ -1725,14 +1715,120 @@ static SDL_GpuGraphicsPipeline* D3D11_CreateGraphicsPipeline(
 		pipeline->vertexStrides = NULL;
 	}
 
-	/* Fragment Shader */
+	/* Resource Layout */
 
-	pipeline->fragmentShader = (ID3D11PixelShader*) fragShaderModule->shader;
-	pipeline->numFragmentSamplers = pipelineCreateInfo->fragmentShaderInfo.samplerBindingCount;
-	pipeline->fragmentUniformBlockSize = D3D11_INTERNAL_NextHighestAlignment(
-		(Uint32) pipelineCreateInfo->fragmentShaderInfo.uniformBufferSize,
-		256
-	);
+    pipeline->resourceLayout.resourceSetCount = pipelineCreateInfo->pipelineResourceLayoutInfo.setLayoutInfoCount;
+    pipeline->resourceLayout.resourceSets = SDL_malloc(
+        pipeline->resourceLayout.resourceSetCount * sizeof(D3D11ShaderResourceSet)
+    );
+
+    for (i = 0; i < pipeline->resourceLayout.resourceSetCount; i += 1)
+    {
+        pipeline->resourceLayout.resourceSets[i].resourceInfoCount =
+            pipelineCreateInfo->pipelineResourceLayoutInfo.setLayoutInfos[i].elementDescriptionCount;
+
+        pipeline->resourceLayout.resourceSets[i].resourceInfos = SDL_malloc(
+            pipeline->resourceLayout.resourceSets[i].resourceInfoCount * sizeof(D3D11ShaderResourceInfo)
+        );
+
+        for (j = 0; j < pipeline->resourceLayout.resourceSets[i].resourceInfoCount; j += 1)
+        {
+            resourceType = pipelineCreateInfo->pipelineResourceLayoutInfo.setLayoutInfos[i].elementDescriptions[j].resourceType;
+            shaderStageFlags = pipelineCreateInfo->pipelineResourceLayoutInfo.setLayoutInfos[i].elementDescriptions[j].shaderStageFlags;
+
+            pipeline->resourceLayout.resourceSets[i].resourceInfos[j].resourceType = resourceType;
+
+            if (shaderStageFlags & SDL_GPU_SHADERSTAGE_VERTEX & SDL_GPU_SHADERSTAGE_FRAGMENT)
```

Comment:
Should this be `if (shaderStageFlags & (SDL_GPU_SHADERSTAGE_VERTEX | SDL_GPU_SHADERSTAGE_FRAGMENT) == ...)` or something?